src/test/java/hudson/plugins/active_directory/docker/TheFlintstonesIT.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, Felix Belzunce Arcos, CloudBees, Inc., and contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.plugins.active_directory.docker;
+
+import static org.junit.Assert.assertEquals;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.naming.NamingException;
+import javax.servlet.ServletException;
+import org.acegisecurity.AuthenticationServiceException;
+import org.acegisecurity.userdetails.UserDetails;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.testcontainers.DockerClientFactory;
+import hudson.plugins.active_directory.ActiveDirectoryDomain;
+import hudson.plugins.active_directory.ActiveDirectorySecurityRealm;
+import hudson.plugins.active_directory.GroupLookupStrategy;
+import hudson.util.Secret;
+
+/**
+ * Integration tests with Docker and requiring custom DNS in the target env with fixed ports.
+ */
+public class TheFlintstonesIT {
+
+    @Rule(order = 0)
```

Comment:
strangely was not done with the old `DockerFixtures`.  So I think this was an existing problem in most other tests in the Jenkins ecosystem where the building of the container counted as part of the timeout in the `JenkinsRule`.